actions/subscribers.go

```diff
@@ -424,39 +426,91 @@ func ExportSubscribers(c *gin.Context) {
 
 	reportSvc := reports.NewReportService(exporters.NewExporter("subscribers"))
 
-	if reportSvc.IsAnotherReportRunning(c, u.ID) {
-		c.JSON(http.StatusForbidden,gin.H{
-			"message": "There is a report already running",
-		})
+	report, err := reportSvc.CreateExportReport(c, u.ID, "subscribers", "")
+	if err != nil {
+		switch err {
+		case reports.ErrAnotherReportRunning:
+			c.JSON(http.StatusForbidden, gin.H{
+				"message": "There is a report already running",
+			})
+		case reports.ErrLimitReached:
+			c.JSON(http.StatusForbidden, gin.H{
+				"message": "You reached the limit",
+			})
+		default:
+			c.JSON(http.StatusUnprocessableEntity, gin.H{
+				"message": "Unable to create export report.",
+			})
+		}
 		return
 	}
 
-	if reportSvc.IsLimitExceeded(c, u.ID) {
-		c.JSON(http.StatusForbidden,gin.H{
-			"message": "You reached the limit",
+	go reportSvc.GenerateExportReport(c.Copy(), report)
```

Comment:
This should return an error, and we should handle it:

```go
go func(ctx context.Context, report entities.Report) {
  err := reportSvc...
  if err != nil {
     ...
  }
}(c.Copy(), report)
```

Reply:
@FilipNikolovski since we will implement endpoint for checking if the report is done. Except logging the error i don't see anything for handling. Do you agree?

Reply:
Yes we can just log the error.